ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,346 @@
+                {
+                    Log.Trace($"BenzingaDataConverter.WriteToFile(): Skipping news article {article.Id} because no tickers are associated with it");
+                    continue;
+                }
+
+                var finalArticleFile = Path.Combine(finalDirectory.FullName, $"{article.Id}.json");
+                var tempArticleFile = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.json");
+                var backupArticleFile = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.json");
+                var articleExists = File.Exists(finalArticleFile);
+
+                Log.Trace($"BenzingaDataConverter.WriteToFile(): Writing article contents to temp: {tempArticleFile}");
+                File.WriteAllText(tempArticleFile, JsonConvert.SerializeObject(article, Newtonsoft.Json.Formatting.None));
+
+                if (articleExists)
+                {
+                    Log.Trace($"BenzingaDataConverter.WriteToFile(): Article already exists. Moving from: {finalArticleFile} - to backup: {backupArticleFile}");
+                    File.Move(finalArticleFile, backupArticleFile);
+                }
+
+                try
+                {
+                    Log.Trace($"BenzingaDataConverter.WriteToFile(): Moving from temp to: {finalArticleFile}");
+                    File.Move(tempArticleFile, finalArticleFile);
+                }
+                catch (Exception e)
+                {
+                    Log.Error(e, $"BenzingaDataConverter.WriteToFile(): Failed to move file to: {finalArticleFile}");
+
+                    if (articleExists)
+                    {
+                        Log.Error($"BenzingaDataConverter.WriteToFile(): Restoring backup article file: {backupArticleFile} to original location: {finalArticleFile}");
+                        File.Move(backupArticleFile, finalArticleFile);
+                    }
+                }
+            }
+
+            // Now compress all of the data we have for a given day
+            var compressedFinal = new FileInfo(Path.Combine(_destinationDirectory.FullName, "content", $"{date:yyyyMMdd}.zip"));
+            var compressedFinalBackup = new FileInfo(Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.zip"));
+
+            if (compressedFinal.Exists)
+            {
+                Log.Trace($"BenzingaDataConverter.WriteToFile(): Moving existing zip file for {date:yyyyMMdd} to temp directory as: {compressedFinalBackup.Name}");
+                compressedFinal.MoveTo(compressedFinalBackup.FullName);
+                compressedFinal.Refresh();
+            }
+
+            try
+            {
+                Log.Trace($"BenzingaDataConverter.WriteToFile(): Compressing {finalDirectory.Name} to {compressedFinal.FullName}");
+                Compression.ZipDirectory(finalDirectory.FullName, compressedFinal.FullName, includeRootInZip: false);
+            }
+            catch (Exception e)
+            {
+                compressedFinal.Refresh();
```

Comment:
oh snap!!! well done -- you can tell I'm doing this line by line. Many people don't know about the `Refresh()` function and it's behavior -- I'm going to leave the comments about it for posterity.